packages/opentelemetry-metrics/src/export/aggregators/Sum.ts

```diff
@@ -14,12 +14,13 @@
  * limitations under the License.
  */
 
-import { Aggregator, Point } from '../types';
+import { Point, Sum, AggregatorKind, SumAggregatorType } from '../types';
 import { HrTime } from '@opentelemetry/api';
 import { hrTime } from '@opentelemetry/core';
 
 /** Basic aggregator which calculates a Sum from individual measurements. */
-export class SumAggregator implements Aggregator {
+export class SumAggregator implements SumAggregatorType {
+  public kind: AggregatorKind.SUM = AggregatorKind.SUM;
```

Comment:
type should not be a value but it should be a type so this
```
public kind: AggregatorKind.SUM = AggregatorKind.SUM;
```
should be this
```
public kind: AggregatorKind = AggregatorKind.SUM;
```
although the type should be already defined by interface which means it should look like this
```
public kind = AggregatorKind.SUM;
```
if you then do this for example
```
public kind = 'foo'
```
You will get a compiler error that kind needs to be a type of `AggregatorKind`

Reply:
Literal value and TypeScript enums can be correctly recognized as a type, which constraints the implementation of the type. In this case, a SumAggregatorType limits their implementation classes to have a field `kind` to be exact `AggregatorKind.SUM`, but not any other `AggregatorKind` value here. So `public kind: AggregatorKind = AggregatorKind.SUM;` will not fit in the expect intention to have a constraint Aggregator definition, and aligning their `toPoint` method to return exactly what they are expected to return.

Reply:
With this PR, a SumAggregator implementation can not have a `kind` field with AggregatorKind.DISTRIBUTION nor `toPoint` returning a Histogram.

Reply:
While taking your comment as a example:
```js
export interface Aggregator {
  kind: AggregatorKind;
  /** Updates the current with the new value. */
  update(value: number): void;

  /** Returns snapshot of the current point (value with timestamp). */
  toPoint(): Point<PointValueType>;
}

export interface Point<T> {
  value: T;
  timestamp: HrTime;
}
```

Such an custom aggregator is a valid one:

```typescript
class CustomAggregator implements Aggregator {
  kind = AggregatorKind.SUM
  update(value:number): void {}
  toPoint(): Point<Distrubution> {
    return { min: 0, max: 0, last: 0, count: 0, sum: 0 };
  }
```

Which is not aligning with the intention of the definition of Aggregator.

Reply:
here you can do it -> https://github.com/open-telemetry/opentelemetry-js/pull/1326,
but your example is wrong the custom aggregator that compiles should look like this

```
class CustomAggregator implements Aggregator {
  kind = AggregatorKind.SUM
  update(value:number): void {}
  toPoint(): Point<Distribution> {
    return { 
      value: {
        min: 0, max: 0, last: 0, count: 0, sum: 0, 
      },
      timestamp: [0,0]
    };
  }
}

```

Reply:
Yes, you are right about the structure of the toPoint return type.

However, the point is that the kind field is not aligned with the toPoint return type, which is the PR intended to fix.

```typescript
class CustomAggregator implements Aggregator {
  kind = AggregatorKind.SUM
  update(value:number): void {}
  toPoint(): Point<Distribution> {
    return { 
      value: {
        min: 0, max: 0, last: 0, count: 0, sum: 0, 
      },
      timestamp: [0,0]
    };
  }
}
```

Reply:
> However, the point is that the kind field is not aligned with the toPoint return type, which is the PR intended to fix.
They cannot be aligned those are 2 different types `kind` is kind of `AggregatorKind` and `Point` is `Point` with dependency of type `AggregatorTypes` where `Distribution` is one of it,  it is just a final aggregator (CustomAggregator) which defines what is what

Reply:
> They cannot be aligned those are 2 different types kind is kind of AggregatorKind and Point is Point with dependency of type AggregatorTypes where Distribution is one of it, it is just a final aggregator (CustomAggregator) which defines what is what

This is exactly what the PR provides. If the CustomAggregator implements `SumAggregatorType` in the PR, it can't have kind with `AggregatorKind.SUM` while the `toPoint` returning a `Point<Distribution>`, since the `SumAggregatorType` declares its `kind` must be `AggregatorKind.SUM` while `toPoint` should returning `Point<Sum>`.